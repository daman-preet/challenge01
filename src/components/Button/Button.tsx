import React from 'react';
import {Button as RNButton, ButtonProps} from 'react-native-ui-lib';

const Button = (props: ButtonProps) => {
  return <RNButton {...props} />;
};

export default Button;
