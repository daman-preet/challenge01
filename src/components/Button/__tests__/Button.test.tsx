import React from 'react';
import * as renderer from 'react-test-renderer';

import Button from '../Button';

it('renders correctly with defaults', () => {
  const button = renderer.create(<Button label={'Sign In'} />).toJSON();
  expect(button).toMatchSnapshot();
});
