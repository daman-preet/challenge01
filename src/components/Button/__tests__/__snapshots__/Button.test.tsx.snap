// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#5A48F5",
      "borderRadius": 999,
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 90,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 9.5,
    }
  }
>
  <Text
    fsTagName="unmask"
    numberOfLines={1}
    style={
      [
        {
          "backgroundColor": "transparent",
          "color": "#20303C",
          "writingDirection": "ltr",
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        [
          {
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          {
            "color": "#FFFFFF",
          },
          undefined,
          undefined,
          undefined,
        ],
      ]
    }
    testID="undefined.label"
  >
    Sign In
  </Text>
</View>
`;
